core/src/main/scala/sttp/tapir/Validator.scala

```diff
@@ -2,29 +2,38 @@ package sttp.tapir
 
 import sttp.tapir.Schema.SName
 import sttp.tapir.macros.ValidatorMacros
+import sttp.tapir.ValidationResult.{Invalid, Valid}
 
 import scala.collection.immutable
 
 sealed trait Validator[T] {
-  def apply(t: T): List[ValidationError[_]]
+
+  def apply(t: T): ValidationResult[T]
 
   def contramap[TT](g: TT => T): Validator[TT] = Validator.Mapped(this, g)
 
   def and(other: Validator[T]): Validator[T] = Validator.all(this, other)
+
   def or(other: Validator[T]): Validator[T] = Validator.any(this, other)
 
-  def show: Option[String] = Validator.show(this)
+  //TODO: Find a better name ?
```

Comment:
what's wrong with `show`? :)

Reply:
Nothing in particular, I just was wondering if there is a more explicit name due to every `show` just define a symbolic representation of the validator like `length<=$value`.  But please, ignore my comment np 😂

Reply:
It's possible that the name could be better, but other classes in tapir use `show` so I'd like to keep things consistent, even if not perfect :)

Reply:
It's clear, makes sense 👍